[13.0][MIG] stock_reception_screen

sebalix opened the pull request.

sebalix committed:
[ADD] New module 'stock_reception_screen'

sebalix committed:
[IMP] stock_reception_screen: switch the screen to a dedicated data model

sebalix committed:
[IMP] stock_reception_screen: handle package, packaging and storage type

sebalix committed:
[IMP] stock_reception_screen: handle package dimensions (height)

sebalix committed:
[IMP] stock_reception_screen: add tests

sebalix committed:
[UPD] README.rst

sebalix committed:
[IMP] stock_reception_screen: update code based on changes in dependencies

sebalix committed:
[IMP] stock_reception_screen: black, isort

sebalix committed:
[MIG] stock_reception_screen: Migration to 13.0

i-vyshnevska commented on stock_reception_screen/models/stock_move.py:
```diff
@@ -0,0 +1,29 @@
+    _inherit = "stock.move"
+
+    reception_screen_current_step = fields.Char(
+        related="picking_id.reception_screen_id.current_step"
```
does current_step of moves have an impact state of the move, in case of done it's ok, but if the screen in "set_pid" step what state move has?

sebalix replied:
I tested this:

1. I start to process the move of a picking, and stop at the `set_lot_number` step for instance (I let the screen opened)
2. In another window I validate the picking as usual, so all its moves are `done`
3. I get back on the reception screen window, and I'm able to continue all the steps (creating a lot, set a expiry date, put the qty received, set the packaging, the PID...). As the screen mainly updates `stock.move.line` there is no error raised by Odoo. But it is an issue indeed... Some checks have to be added at each step I think to raise an error is the move is already processed (cancel or done).
ping @jgrandguillaume

i-vyshnevska commented on stock_reception_screen/models/stock_move.py:
```diff
@@ -0,0 +1,29 @@
+# Copyright 2019 Camptocamp SA
```
```suggestion
# Copyright 2020 Camptocamp SA
```
all over the code

i-vyshnevska commented on stock_reception_screen/models/stock_move.py:
```diff
@@ -0,0 +1,29 @@
+    def action_select_move(self):
+        """Set the move as the current one at the picking level."""
+        self.ensure_one()
+        # self.picking_id.reception_screen_id.button_reset()
```
```suggestion
```
?

i-vyshnevska commented on stock_reception_screen/models/stock_reception_screen.py:
```diff
@@ -0,0 +1,528 @@
+
+    def _before_select_move_to_set_lot_number(self):
+        """Decide if we have to handle lots on the current move."""
+        return self.current_move_id.has_tracking != "none"
```
we don't have different handlings for serial tracking and lots here?
in meaning, if the lot has tracking by serial we can constrain user with qty == 1, before sending data to ORM

gurneyalex commented:
Travis is failing because stock_packaging_dimension is not available. 

@sebalix  Can you add the OCA/product-attributes repositories to the oca_dependencies.txt file (cf https://github.com/OCA/wms/blob/12.0/oca_dependencies.txt)

grindtildeath commented on stock_reception_screen/__manifest__.py:
```diff
@@ -0,0 +1,33 @@
+        # OCA/stock-logistics-workflow
+        "stock_quant_package_dimension",
+        # OCA/wms
+        "stock_storage_type_putaway_strategy",
```
```suggestion
        "stock_storage_type",
```

sebalix committed:
Enable 'setuptools-odoo-make-default' pre-commit hook

sebalix committed:
[IMP] stock_reception_screen: filter destination locations on sub-locations only

sebalix committed:
[IMP] stock_reception_screen: do not ask for Height when pallet product packaging is used

sebalix committed:
[IMP] stock_reception_screen: black, isort, prettier

sebalix committed:
[IMP] stock_reception_screen: warning alert if the lot received is older than the existing one

sebalix committed:
[IMP] stock_reception_screen: blocking alert when a product doesn't have a proper information filled on the product packaging

sebalix committed:
[FIX] stock_reception_screen: ensure that splitted moves are assigned

+ do not block when receiving pallets with missing pkg info

Tonow-c2c commented:
your PR is one [MIG] but we don't have the module history :disappointed_relieved:

jbaudoux commented on stock_reception_screen/models/stock_reception_screen.py:
```diff
@@ -532,6 +532,13 @@ def _check_package_data(self):
             self.env.user.notify_warning(message="", title=msg)
             return False
+        if self.current_move_line_product_packaging_id and (
+            not self.current_move_line_product_packaging_id.volume
```
The volume is in m3 and when the packaging is small, it equals to 0 after roundnig. Can you check instead the length/width/height that are in mm ?

jgrandguillaume commented:
@sebalix Can you check travis ?

sebalix committed:
[FIX] stock_reception_screen: fix runbot

sebalix committed:
[IMP] stock_reception_screen: rename 'PID' terms to 'Package'

sebalix committed:
[FIX] stock_reception_screen: show 'Reception' button only for relevant pickings on tree view

sebalix committed:
[FIX] stock_reception_screen: fix set_destination step

guewen commented:
@sebalix FYI I rebased your PR to solve a conflict with oca-dependencies.txt

guewen commented:
@sebalix the title says "WIP", still true?

TDu committed:
[IMP] stock_reception_screen

When a stock move is received in multiple steps, the lot id has to be
entered everytime, which is a lost of time.
Here we keep the last lot_id used for a move and it is set as default
for the next move line to be received.
It can always be changed if needed.

Improve the flow of screens, so that when a product has been revceived
if there is more of the same product to receive open directely the next
move without going through the list.

Fix the reset button.

Add the vendor code when receiving a product.

sebalix committed:
[FIX] stock_reception_screen: reset lot life date

sebalix committed:
[IMP] stock_reception_screen: height management

The height of the package was required if the product packaging was a
pallet-type. Now the height is required is the storage type is
configured this way (could be used outside of "Pallet" kind of storage
types), this check doesn't depend on product packaging anymore.

sebalix commented:
Last commit 972bba069c775f53e951d2235bfbb10ed7bcd312 depends on https://github.com/OCA/wms/pull/50

@guewen Once #50 is merged I suppose this one is ready for merge too.
(that said, I am not satisfy of this module, a bit hackish I think)

jgrandguillaume reviewed (approved):
LGTM, functional test

jgrandguillaume commented:
/ocabot merge nobump

sebalix commented:
Merge command needs to be relaunch, build failed due to Ubuntu APT repository unavailable:
```
Err:21 http://archive.ubuntu.com/ubuntu xenial InRelease

  Could not connect to archive.ubuntu.com:80 (91.189.88.142), connection timed out [IP: 91.189.88.142 80]

Err:22 http://archive.ubuntu.com/ubuntu xenial-updates InRelease

  Unable to connect to archive.ubuntu.com:http: [IP: 91.189.88.142 80]

Err:23 http://archive.ubuntu.com/ubuntu xenial-backports InRelease

  Unable to connect to archive.ubuntu.com:http: [IP: 91.189.88.142 80]

Fetched 5,231 kB in 60s (87.1 kB/s)

Reading package lists...

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial/InRelease  Could not connect to archive.ubuntu.com:80 (91.189.88.142), connection timed out [IP: 91.189.88.142 80]

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial-updates/InRelease  Unable to connect to archive.ubuntu.com:http: [IP: 91.189.88.142 80]

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial-backports/InRelease  Unable to connect to archive.ubuntu.com:http: [IP: 91.189.88.142 80]

W: Some index files failed to download. They have been ignored, or old ones used instead.

The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends $(travis_apt_get_options) install expect-dev" failed and exited with 100 during .
```

TDu committed:
Add supplier reference on move list

TDu committed:
Add storage type name in packaging list

TDu committed:
Reduce screen real estate used

TDu committed:
Add custom autofocus

TDu committed:
Fix linting

sebalix committed:
[FIX] stock_reception_screen: fix package data auto-complete

The package data (product packaging, package storage type and height
here) are now handled this way:

1) first these data are auto-completed on the screen to let the user
   changing them if needed, but they are not directly set on the package
   (method `_autocomplete_package_data`)
2) once the user confirm the "select_packaging" step, these data will be
   set on the related package (method `_set_package_data`)

This avoids some problems related to existing constraints on the package,
like the height which can be mandatory depending on the storage type.

jgrandguillaume commented:
/ocabot merge nobump
